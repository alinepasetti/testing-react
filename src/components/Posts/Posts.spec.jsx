import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Posts } from ".";

const MOCK_PROPS = [
  {
    title: "title 1",
    id: 1,
    body: "body",
    image: {
      url: "url.com",
      title: "image title",
    },
  },
  {
    title: "title 2",
    id: 2,
    body: "body",
    image: {
      url: "url.com",
      title: "image title",
    },
  },
];

describe("<Posts />", () => {
  it("render the button with the text", () => {
    render(<Posts posts={MOCK_PROPS} />);
    const headings = screen.getAllByRole("heading", { name: /title/i });
    expect(headings).toHaveLength(2);
    const images = screen.getAllByRole("img", { name: "image title" });
    expect(images).toHaveLength(2);
    const bodies = screen.getAllByText("body");
    expect(bodies).toHaveLength(2);
  });

  it("should match snapshot", () => {
    const postsElement = renderer.create(<Posts posts={MOCK_PROPS} />).toJSON();
    expect(postsElement).toMatchSnapshot();
  });
});
