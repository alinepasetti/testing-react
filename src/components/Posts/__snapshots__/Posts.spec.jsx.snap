// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  className="posts"
>
  <div
    className="post"
  >
    <img
      alt="image title"
      className="post-image"
      src="url.com"
    />
    <div
      className="post-content"
    >
      <h2>
        1
         
        title 1
      </h2>
      <p>
        body
      </p>
    </div>
  </div>
  <div
    className="post"
  >
    <img
      alt="image title"
      className="post-image"
      src="url.com"
    />
    <div
      className="post-content"
    >
      <h2>
        2
         
        title 2
      </h2>
      <p>
        body
      </p>
    </div>
  </div>
</div>
`;
